src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_CAPACITY = 8;
+    private static final float MULTIPLIER = 1.5f;
+
+    private T[] data;
+    private int size;
+
+    public MateStack() {
+        data = (T[]) new Object[INITIAL_CAPACITY];
+    }
 
+    public void push(T value) {
+        if (size == data.length) {
+            resize();
+        }
+        data[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return data[size - 1];
     }
 
     public T pop() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        final T element = data[size - 1];
+        System.arraycopy(data, 0, data, 0, size - 1);
+        size--;
+        return element;
```

Comment:
You don't need to copy array each time you delete element, work with what you have.